docs/design/SYSTEM_DESIGN.md

```diff
@@ -6,101 +6,138 @@
 ### Non-Goals
 * Create a detailed design and interface specification for each system component.
 
+### Terminology
+* Control Plane- A collection of inter-related software components for providing application gateway and routing
+  functionality. The control plane is implemented by Envoy Gateway and provides services for managing the data plane.
+  These services are detailed in the [components](#components) section.
+* Data Plane- Provides intelligent application-level traffic routing and is implemented by Envoy proxy. Envoy Gateway
+  manages the data plane as one or more Envoy proxies.
```

Comment:
Small nit. In the "Control Plane" definition (which is right on), there is a mention of "[...] provides services for managing the data plane". Because of this mention, I'd remove the superfluous sentence "Envoy Gateway manages the data plane as one or more Envoy proxies." since it repeats known information and loops back to the previous definition.